### PR TITLE
miscellaneous refactorings and bug fixes

### PR DESCRIPTION
**Pull request by aspiers**
Since mkcloud isn't (yet) gated on itself (BTW it should be),
this needs more testing before merge, but it's ready for review.

### CODE REVIEW

**Comment by dirkmueller** [comment]
@aspiers Generally I think those changes are okay, some of the commits should be squashed imho though. 

I'd appreciate if you could prefix them with "mkcloud:" as that seems to be the general style of git commit messages here.

**Comment by aspiers** [comment]
> @aspiers Generally I think those changes are okay, some of the commits should be squashed imho though.

Which ones?

> I'd appreciate if you could prefix them with "mkcloud:" as that seems to be the general style of git commit messages here.

Sure, will do.  Thanks for the review!

**Comment by aspiers** [comment]
@dirkmueller Updated according to your feedback.

**Comment by dirkmueller** [comment]
@aspiers: The "only delete if X exist" style commits are all related imho, which can be summarized under "remove noisy output" and could be squashed together imho.

**Comment by aspiers** [comment]
Rebased.

@dirkmueller Commits can be logically distinct whilst still being related to other commits.  I see your point but in this case I think our personal tastes differ.  Regardless, I tried the squash to keep you happy, but it is not trivial because one of the commits depends on a commit unrelated to the others.  After 5 minutes of merge conflicts I gave up because I couldn't see any benefits which justify the extra work.  If there's something I'm missing please let me know and I can try again.

**Comment by aspiers** [comment]
Rebased.  Please can we merge this ASAP?  It's been in review for a while now and I have lots of other changes which are blocked by it.

File: scripts/mkcloud [hunk 1]
```diff
@@ -359,14 +415,15 @@ function onhost_deploy_admin_image()
 {
     pushd /tmp
     wget --progress=dot:mega -nc http://clouddata.cloud.suse.de/images/SP3-64up.qcow2
-    qemu-img convert -p SP3-64up.qcow2 /dev/$cloudvg/$cloud.admin
+
+    qemu-img convert -p SP3-64up.qcow2 $admin_node_disk
     popd
 
-    if fdisk -l /dev/$cloudvg/$cloud.admin | grep -q 2056192 ; then
+    if fdisk -l $admin_disk | grep -q 2056192 ; then
```
Comment (bmwiedemann) [inline scripts/mkcloud]: $admin_disk should be $admin_node_disk

Comment (aspiers) [inline scripts/mkcloud]: Great catch, thanks!  And also the change was in the wrong commit.  Now fixed.

**Pull request merged**